maven-scm-providers/maven-scm-providers-git/maven-scm-provider-jgit/src/test/java/org/apache/maven/scm/provider/git/jgit/command/checkout/JGitSshCheckOutCommandTckTest.java

```diff
@@ -0,0 +1,149 @@
+import org.eclipse.jgit.internal.transport.sshd.OpenSshServerKeyDatabase;
+import org.eclipse.jgit.transport.CredentialsProvider;
+import org.eclipse.jgit.transport.sshd.ServerKeyDatabase;
+import org.eclipse.jgit.util.FileUtils;
+import org.junit.Test;
+import org.slf4j.Logger;
+
+/** @author <a href="mailto:struberg@yahoo.de">Mark Struberg</a> */
+public class JGitSshCheckOutCommandTckTest
+    extends GitSshCheckOutCommandTckTest
+{
+
+    public JGitSshCheckOutCommandTckTest() throws GeneralSecurityException, IOException
+    {
+        super();
+    }
+
+    @Override
+    protected String getScmProvider()
+    {
+        return "jgit";
+    }
+
+    @Override
+    public void initRepo() throws Exception
+    {
+        super.initRepo();
+        JGitTestScmProvider provider = (JGitTestScmProvider) getScmManager().getProviderByRepository( getScmRepository() );
+        // accept all hosts
+        provider.registerCheckOutCommandCallback( new Consumer<JGitCheckOutCommand>()
+        {
+            @Override
+            public void accept( JGitCheckOutCommand command )
+            {
+                command.setSshSessionFactorySupplier( AcceptAllHostsSshdSessionFactory::new );
+            }
+
+        } );
+    }
+
+    private static final class AcceptAllHostsSshdSessionFactory extends ScmProviderAwareSshdSessionFactory
+    {
+        public AcceptAllHostsSshdSessionFactory( GitScmProviderRepository repo, Logger logger )
+        {
+            super( repo, logger );
+        }
+
+        @Override
+        protected ServerKeyDatabase createServerKeyDatabase( File homeDir, File sshDir )
+        {
+            return new OpenSshServerKeyDatabase( false, null )
+            {
+                @Override
+                public boolean accept( @NonNull String connectAddress,
+                        @NonNull InetSocketAddress remoteAddress,
+                        @NonNull PublicKey serverKey,
+                        @NonNull Configuration config, CredentialsProvider provider )
+                {
+                    return true;
+                }
+            };
+        }
+    }
+
+    @Override
+    protected void deleteDirectory( File directory )
+        throws IOException
+    {
+        if ( directory.exists() )
+        {
+            FileUtils.delete( directory, FileUtils.RECURSIVE | FileUtils.RETRY );
+        }
+    }
+
+    @Test
+    public void testCheckOutCommandTestWithOpenSSHPrivateKey()
+        throws Exception
+    {
+        configureKeypairFromResource( getScmRepository(), "sample_rsa", "mySecret");
+        super.testCheckOutCommandTest();
+    }
+
+    private void configureKeypairFromResource( ScmRepository repository, String resourceName, String passphrase )
+        throws IOException, GeneralSecurityException
+    {
+        // accept public key
+        try ( InputStream publicKeyInputStream = this.getClass().getResourceAsStream( "/ssh-keypairs/" + resourceName + ".pub"))
```

Comment:
spaces missing

Reply:
Done in https://github.com/apache/maven-scm/pull/155/commits/9f86604ea544ad802dffcd48e5c381facebb9269.